public/app/features/dashboard/dashgrid/DashboardEmpty/DashboardEmpty.tsx

```diff
@@ -29,99 +30,151 @@ interface InternalProps {
 const InternalDashboardEmpty = ({ onAddVisualization, onAddLibraryPanel, onImportDashboard }: InternalProps) => {
   const styles = useStyles2(getStyles);
 
-  const [searchParams] = useSearchParams();
+  const [searchParams, setSearchParams] = useSearchParams();
   const dashboardLibraryDatasourceUid = searchParams.get('dashboardLibraryDatasourceUid');
+  const showLibraryModal = searchParams.get('dashboardLibraryModal') === 'open';
+
+  // Validate and get default tab from URL params
+  const tabParam = searchParams.get('dashboardLibraryTab');
+  const defaultTab: 'datasource' | 'community' = tabParam === 'community' ? 'community' : 'datasource';
```

Comment:
why does the `DashboarEmpty` component have to know about the implementation of `DashboardLibraryModal`?
I would put all this logic inside `DashboardLibraryModal` so it's independent and also there's cohesion / are decoupled. 
This is also good if we want to reuse this behavior and render the modal on another page in the future

Reply:
I think there was a misunderstanding with your old comment, I thought you were referring to moving this logic to the `Modal`.